caddyhttp/browse/browse.go

```diff
@@ -291,18 +293,13 @@ func isSymlinkTargetDir(f os.FileInfo, urlPath string, config *Config) bool {
 	if !isSymlink(f) {
 		return false
 	}
-	fullPath := func(fileName string) string {
-		fullPath := filepath.Join(string(config.Fs.Root.(http.Dir)), urlPath, fileName)
-		return filepath.Clean(fullPath)
-	}
-	target, err := os.Readlink(fullPath(f.Name()))
-	if err != nil {
-		return false
-	}
-	targetInfo, err := os.Lstat(fullPath(target))
+
+	fullPath := filepath.Join(string(config.Fs.Root.(http.Dir)), urlPath, f.Name())
+	targetInfo, err := os.Stat(fullPath)
```

Comment:
`fullPath` function and `os.Readlink` should not be needed anymore. When we use `os.Stat` instead of `os.Lstat` the OS will take care resolving the symlink and stat what it points to.

I get a bit concern that `os.*` functions are used here without cleaning `urlPath` (seems to under total user control). Original code cleaned fullPath but then it's too late i think. Maybe we can stat thru `config.Fs.Root.Open(..).Stat()` somehow?